pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1408,6 +1421,11 @@ public void updateIndexingConfigFor(String tableName, TableType type, IndexingCo
     setExistingTableConfig(tableConfig, tableNameWithType, type);
 
     if (type == TableType.REALTIME) {
+      // Check if HLC table is allowed
```

Comment:
Did you look into the possibility of having a single validateIndexingConfig method?

Reply:
Added.